packages/imperative/src/logger/src/log4jsToWinston.ts

```diff
@@ -0,0 +1,273 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { transports, format, LoggerOptions, addColors } from "winston";
+import * as os from "os";
+import * as dayjs from "dayjs";
+
+export const customLevels = {
+    levels: {
+        fatal: 0,
+        error: 1,
+        warn: 2,
+        info: 3,
+        mark: 4,
+        debug: 5,
+        trace: 6,
+        all: 7
+    },
+    colors: {
+        fatal: "magenta",
+        error: "red",
+        warn: "yellow",
+        info: "green",
+        mark: "white",
+        debug: "blue",
+        trace: "gray",
+    },
+};
+
+addColors(customLevels.colors);
+
+/**
+ * Translates a log4js pattern string into a Winston format object.
+ * Handles common tokens like %d, %p, %m, %c, %n, %h, %z, %%.
+ * Supports multiple %d tokens with different format specifiers (e.g., %d{yyyy/MM/dd}, %d{hh:mm:ss}).
+ * Strips color codes (%[ and %]).
+ * @param pattern The log4js pattern string.
+ * @returns A Winston format object.
+ */
+export function translateLog4jsPattern(
+    pattern: string
+): ReturnType<typeof format.combine> {
+    // Mapping from log4js date format specifiers to dayjs format specifiers
+    // https://github.com/log4js-node/log4js-node/issues/1012#issuecomment-1017144708
+    const mapLog4jsToDayjsFormat = (log4jsFormat: string): string => {
```

Comment:
Should we just export this function (so that we don't have to define it every time `translateLog4jsPattern` is called?

Reply:
This was more of a C++ habit of mine where I like to use lambdas if I don't reuse the logic elsewhere 😋 
But we can definitely expose this inner function's logic in case an extender needs it.

Plan to address in an upcoming commit 👍